File: gis-mock-frontend/src/interfaces/PolygonDTO.ts
// src/interfaces/PolygonDTO.ts

/**
 * Represents a saved polygon and its related real estate objects.
 */
export interface PolygonDTO {
    id: string;
    name: string;
    // The set of latitude/longitude pairs that define the polygon boundary
    coordinates: Array<{ lat: number; lng: number }>;

    // List of real estate objects that were inside this polygon
    realEstateObjects: Array<string>; // You could store just IDs or RealEstate[] with details

    // Timestamps for display
    dateCreated: string;
    dateUpdated: string;
}

File: gis-mock-api/src/main/java/com/example/realestate/dto/model/PolygonDto.java
package com.example.realestate.dto.model;

import lombok.Data;

import java.util.List;

@Data
public class PolygonDto {
    private String id;
    private String name;
    private List<CoordinateDto> coordinates;
    private List<String> realEstateObjects;
    private String dateCreated;
    private String dateUpdated;

    private String arcgisLayerId;
    private String arcgisPolygonId;
}
